Test suite: Hold the job for cancelling

zdohnal opened the pull request:
On some quicker machines the print-job operation finishes quicker than
we can cancel it in the test - so add job-hold-until job template
attribute, which will hold the job until it is cancelled by the next
test.

The next test passes because it has both statuses (correct and error one)
as expected, but cupsd generates an additional error in log if the job
cancelling fails, which would be more dirty to cover in run-stp-test.sh.

Fixes? #155

zdohnal commented:
@shirelyhhdl @sp1ritCS

michaelrsweet reviewed (changes requested):
The only problem with this change is that the "ipp-1.1.test" file is also meant to test real printers, many of which will not support job-hold-until. So the use of job-hold-until should be made conditional on whether job-hold-until is supported.

zdohnal commented:
@michaelrsweet I haven't found out how to make a specific attribute optional, so I've added a new print job with job-hold-until if the printer supports it, and adjusted requests count in run-stp-tests.sh accordingly.

sp1ritCS commented:
I don't know how soon I'm able to test it, but I'll try

zdohnal commented:
@sp1ritCS @shirelyhhdl any updates?
